src/Fixer/Strict/StrictTypesFixer.php

```diff
@@ -0,0 +1,170 @@
+    {
+        return count($tokens) > 1 && $tokens[0]->isGivenKind(T_OPEN_TAG);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // check for if declaration already in file
+        $sequence = $this->getDeclareStrictTypeSequence();
+        $searchIndex = $tokens->getNextMeaningfulToken(0);
+        if (null === $searchIndex) {
+            $this->insertNewSequence($tokens); // not found, insert declare statement
+
+            return;
+        }
+
+        $sequenceLocation = $tokens->findSequence($sequence, $searchIndex, null, false);
+        if (null === $sequenceLocation) {
+            $this->insertNewSequence($tokens); // not found, insert declare statement
+
+            return;
+        }
+
+        // found, fix casing
+        $this->fixSequenceCasing($sequenceLocation);
+
+        end($sequenceLocation);
+        $sequenceEndIndex = $tokens->getNextMeaningfulToken(key($sequenceLocation));
+        if ($tokens[$sequenceEndIndex]->isGivenKind(T_CLOSE_TAG)) {
+            return; // don't move, comment placement might cause invalid code if moved
+        }
+
+        // check if the declaration is at the right location
+        reset($sequenceLocation);
+        $sequenceStartIndex = key($sequenceLocation);
+        if (1 === $sequenceStartIndex) {
+            // already at the correct location
+            $this->fixSpaceAroundSequence($tokens, $sequenceStartIndex, $sequenceEndIndex);
+
+            return;
+        }
+
+        $sequenceLocation[$sequenceEndIndex] = $tokens[$sequenceEndIndex];
+        $this->moveSequence($tokens, $sequenceLocation);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Force strict types declaration in all files.';
+    }
+
+    /**
+     * @param Token[] $sequenceLocation
+     */
+    private function fixSequenceCasing(array $sequenceLocation)
+    {
+        foreach ($sequenceLocation as $token) {
+            $token->setContent(strtolower($token->getContent()));
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $startIndex
+     * @param int    $endIndex
+     */
+    private function fixSpaceAroundSequence(Tokens $tokens, $startIndex, $endIndex)
+    {
+        if ($tokens[1 + $endIndex]->isWhitespace()) {
+            $content = $tokens[1 + $endIndex]->getContent();
+            if (false === strpos($content, "\n")) {
+                $tokens[1 + $endIndex]->setContent("\n".ltrim($content));
+            }
+        } else {
+            $tokens->insertAt(1 + $endIndex, new Token(array(T_WHITESPACE, "\n")));
+        }
+
+        // transform "<?php\n" to "<?php " if needed
+        if (false !== strpos($tokens[0]->getContent(), "\n")) {
+            $tokens[0]->setContent(trim($tokens[0]->getContent()).' ');
+        } elseif ($tokens[1]->isWhitespace()) {
+            $tokens[1]->setContent(' ');
```

Comment:
I think we should remove the whitespace token to get `<?php declare..` (first space belongs to open tag token).